### PR TITLE
Added Whitelist of e-mail domains

### PR DESCRIPTION
**Pull request by cydh**
* Added config `UseWhitelistEmailDomains` and its file config/whitelistemaildomains.php
* Changed `AllowDuplicateEmails` from boolean to integer to allowed duplicate e-mail by n numbers. 0: No duplicate allowed, -1: ignore duplication check

- [ ] Community test
- [x] Add populer domains in whitelistemaildomains.php

### CODE REVIEW

**Comment by cydh** [comment]
This is a neat list from nice project https://github.com/mailcheck/mailcheck/wiki/List-of-Popular-Domains

**Comment by hurtsky** [comment]
why not implement the mailcheck.js xD

File: lib/Flux/LoginServer.php [hunk 1]
```diff
@@ -178,13 +178,29 @@ public function register($username, $password, $confirmPassword, $email,$email2,
 			throw new Flux_RegisterError('Username is already taken', Flux_RegisterError::USERNAME_ALREADY_TAKEN);
 		}
 		
-		if (!Flux::config('AllowDuplicateEmails')) {
-			$sql = "SELECT email FROM {$this->loginDatabase}.login WHERE email = ? LIMIT 1";
+		if (Flux::config('UseWhitelistEmailDomains') && Flux::config('WhitelistEmailDomains')) {
+			$domains = Flux::config('WhitelistEmailDomains')->toArray();
+			$i = 0;
+			echo $email_str[2];
```
Comment (secretdataz) [inline lib/Flux/LoginServer.php]: is this intentional?

Comment (cydh) [reply lib/Flux/LoginServer.php]: not
```suggestion
```

**Comment by Everade** [comment]
This would block out all personal domains, and the current list is still lacking a very substantional amount of popular domains.
I think a blacklist would make more sense to get at least rid of trashmails, or a service implementation that offers more settings.

**Comment by Akkarinage** [comment]
I agree, a blacklist would make more sense in this scenario.

**Pull request closed without merging**